### ISSUE TITLE
`weak.isWeakRef()` returns `true` when given a promise.

### ISSUE THREAD

**Issue by danfuzz** [TooTallNate/node-weak#83]
I have code that looks like this:

```javascript
    const already = this._complexes.get(docId);
    if (already) {
      // There's something in the cache...
      if (weak.isWeakRef(already)) {
        // It's a weak reference. If not dead, it refers to a `FileComplex`.
        if (!weak.isDead(already)) {
          const result = weak.get(already);
          result.log.info('Retrieved cached complex.');
          return result;
        }
        // else, it's a dead weak reference. We'll fall through and construct a
        // new result.
        log.info(`[${docId}] Cached complex was gc'ed.`, already);
      }
    }
```

(extracted from <https://github.com/danfuzz/bayou/blob/master/local-modules/doc-server/DocServer.js>)

I have seen cases in my logs where that last `log` line has printed a log, and `already` — which should be a weak ref given that it's inside `if (weak.isWeakRef(already) {...}` — is in fact an instance of `Promise`. (More specifically, it's a pending promise which will ultimately resolve to something that is _not_ going to be a weak reference.) This does not happen consistently; sometimes promises will be correctly determined not to be weak refs.

I noticed this on Node 8.1.4 (macOS Sierra, installed via `brew`). I upgraded Node to 8.9.1, and I haven't seen the problem resurface yet. Nonetheless I thought this might be of interest. Notably, I didn't see anything that looked salient in the Node CHANGELOG. (But I didn't check the V8 CHANGELOG.)

**Comment by danfuzz** [comment]
Update: Saw it happen on Node 8.9.1

**Comment by danfuzz** [comment]
BTW I'm changing my code to avoid calling `isWeakRef()` (instead testing if the result is a promise, which I'm hoping will be more reliable).

**Comment by danfuzz** [comment]
**Update**

After paring down my test case, I figured out that this is 100% consistent. `weak.isWeakRef()` seems to _always_ return `true` given a promise:

```javascript
const weak = require('weak');
const p = new Promise((res, rej) => {});

console.log(weak.isWeakRef(p));
```

```
$ npm install weak
$ node --version
v8.4.0
$ node index.js
true
```

I'm updating the issue title accordingly.

**Comment by danfuzz** [comment]
After a brief jaunt through the source, I found <https://github.com/TooTallNate/node-weak/blob/master/src/weakref.cc#L184>:

```c++
bool isWeakRef (Local<Value> val) {
  return val->IsObject() && val.As<Object>()->InternalFieldCount() == 1;
}
```

I guess promises also have an internal field count of `1`.

**Comment by TooTallNate** [comment]
Fixed by #88.
